### PR TITLE
Refactor: parse whole tx before signing

### PR DESCRIPTION
**Pull request by ppershing**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -10,9 +10,9 @@ const BASE58_ALPHABET =
   "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
 const bs58 = basex(BASE58_ALPHABET);
 
-export const MAX_UINT_64_STR = "18446744073709551615";
+//export const MAX_UINT_64_STR = "18446744073709551615";
```
Comment (refi93) [inline src/utils.ts:13]: probably forgotten commented out constants after moving them to the parsing module

File: src/interactions/signTx.ts [hunk 2]
```diff
@@ -498,134 +480,115 @@ const signTx_getWitness = async (
...
-  );
 
-  const appHasStakePoolOwnerSupport = await isLedgerAppVersionAtLeast(
-    _send,
-    2,
-    1
-  );
-  if (isSigningPoolRegistrationAsOwner && !appHasStakePoolOwnerSupport) {
-    throw Error(Errors.INCORRECT_APP_VERSION);
+function generateWitnessList(tx: ParsedTransaction): ValidBIP32Path[] {
```
Comment (refi93) [inline src/interactions/signTx.ts:484]: as this returns the list of paths, not the witnesses themselves, I see this rather as `getPathsToWitness`

File: src/interactions/signTx.ts [hunk 3]
```diff
@@ -86,42 +68,42 @@ const signTx_init = async (
...
-    _serializePoolRegistrationCode(flags.isSigningPoolRegistrationAsOwner),
-    uint32_to_buf(numInputs),
-    uint32_to_buf(numOutputs),
-    uint32_to_buf(numCertificates),
-    uint32_to_buf(numWithdrawals),
-    uint32_to_buf(numWitnesses),
+    _serializePoolRegistrationCode(tx.isSigningPoolRegistrationAsOwner),
+    uint32_to_buf(tx.inputs.length as Uint32_t),
+    uint32_to_buf(tx.outputs.length as Uint32_t),
+    uint32_to_buf(tx.certificates.length as Uint8_t),
```
Comment (refi93) [inline src/interactions/signTx.ts:104]: shouldn't this be a casting to Uint32_t as well?

Comment (ppershing) [reply src/interactions/signTx.ts:104]: should

**Pull request merged**